src/script/components/asset/controls/MediaButton.tsx

```diff
@@ -0,0 +1,105 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import cx from 'classnames';
+import ko from 'knockout';
+import AssetLoader from 'Components/asset/AssetLoader';
+import React, {useEffect, useState} from 'react';
+import type {FileAsset} from '../../../entity/message/FileAsset';
+import {AssetTransferState} from '../../../assets/AssetTransferState';
+import {noop} from 'Util/util';
+import {registerReactComponent} from 'Util/ComponentUtil';
+
+export interface MediaButtonProps {
+  asset: FileAsset;
+  cancel?: () => void;
+  large: boolean;
+  pause?: () => void;
+  play: () => void;
+  src: HTMLMediaElement;
+  transferState: AssetTransferState;
+  uploadProgress: number;
+}
+
+const MediaButton: React.FC<MediaButtonProps> = ({
+  src,
+  large,
+  asset,
+  uploadProgress,
+  transferState,
+  play,
+  pause = noop,
+  cancel = noop,
+}) => {
+  const [isPlaying, setIsPlaying] = useState<boolean>(false);
+  const onPlay = () => setIsPlaying(true);
+  const onPause = () => setIsPlaying(false);
+
+  const mediaElement = src;
+
+  useEffect(() => {
+    mediaElement.addEventListener('playing', onPlay);
+    mediaElement.addEventListener('pause', onPause);
+
+    return () => {
+      mediaElement.removeEventListener('playing', onPlay);
+      mediaElement.removeEventListener('pause', onPause);
+    };
+  }, [mediaElement]);
+
+  const isUploaded = transferState === AssetTransferState.UPLOADED;
+  const isDownloading = transferState === AssetTransferState.DOWNLOADING;
+  const isUploading = transferState === AssetTransferState.UPLOADING;
+
+  const mediaButtonPlay = (
+    <div className="media-button media-button-play icon-play" onClick={play} data-uie-name="do-play-media"></div>
+  );
+
+  const mediaButtonPause = (
+    <div className="media-button media-button-pause icon-pause" onClick={pause} data-uie-name="do-pause-media"></div>
+  );
+
+  const isDownloadingIndicator = (
+    <AssetLoader large={large} loadProgress={ko.unwrap(asset.downloadProgress)} onCancel={cancel} />
+  );
+
+  const isUploadingIndicator = <AssetLoader large={large} loadProgress={uploadProgress} onCancel={cancel} />;
+
+  return (
+    <div
+      className={cx({
+        'media-button-lg': large,
+      })}
+    >
+      {isUploaded && !isPlaying && mediaButtonPlay}
+      {isUploaded && isPlaying && mediaButtonPause}
+      {isDownloading && isDownloadingIndicator}
+      {isUploading && isUploadingIndicator}
+    </div>
+  );
+};
+
+export default MediaButton;
+
+registerReactComponent<MediaButtonProps>('media-button', {
```

Comment:
When do we need to add the props (`MediaButtonProps`) and when it's obsolete? /cc @ffflorian

Reply:
It's actually not obsolete. It's optional but advisable. As a rule: Always use it when you declare `optionalParams`. But I like to always use it because it only gets better with it. :)